### DISCUSSION TITLE
How to catch UserRejectedRequestError?

### DISCUSSION THREAD

**Discussion by apostgit** [rainbow-me/rainbowkit#1444] [Q&A]
I'm using `openConnectModal` from `useConnectModal` to open the connect modal. When for example you press on Metamask and cancel the connection it throws an error. How can we catch that to make the UI respond to the error?

Thanks in advance everyone! 😄

**Comment by DanielSinclair** [comment]
@apostgit Are you still experiencing this issue? Do you mean an exception is thrown, or a `console.error()` is printed? Unfortunately we do see console errors from underlying WalletConnect libraries that are out of our control.

**Comment by amonsosanz** [comment]
I'm having this same issue. I can't find a way to capture the `UserRejectedRequestError` error using `openConnectModal` and MetaMask.

I see this in the console when I cancel the connection request from MetaMask:

<img width="634" alt="image" src="https://github.com/user-attachments/assets/616a5581-8ac8-43eb-9391-7c7d69929a29" />

But how can I capture this error to react from the UI?

The error from Wagmi's `useConnect` is also `null` after the user rejects the connection.

<img width="457" alt="image" src="https://github.com/user-attachments/assets/e85a215e-c669-45b7-b668-e76c8cb6a95e" />
